docs/how-to/profiling.txt

```diff
@@ -0,0 +1,98 @@
+.. _howto-profiling:
+
+###############################
+ How To: Profile An Experiment
+###############################
+
+When optimizing the training speed of a model, the first step is to
+understand where and why training is slow. Once the bottlenecks have
+been identified, the next step is to do further investigation and
+experimentation to alleviate those bottlenecks.
+
+To understand the performance profile of a training job, the training
+code and infrastructure needs to be instrumented. There are many
+different layers that can be instrumented, from raw throughput all the
+way down to GPU kernels.
+
+Determined provides two tools out-of-the-box for instrumenting training:
+
+-  :ref:`System Metrics <howto-profiling-system-metrics>`: measurements
+   of hardware usage
+-  :ref:`Timings <howto-profiling-timings>`: durations of actions taken
+   during training, such as dataloading
+
+System Metrics are useful to see if the software is taking full
+advantage of the available hardware, particularly around GPU usage,
+dataloading, and network communication during distributed training.
+Timings are useful for identifying the section of code to focus on for
+optimizations. Most commonly, Timings help answers the question of
+whether the dataloader is the main bottleneck in training.
+
+.. _howto-profiling-system-metrics:
+
+****************
+ System Metrics
+****************
+
+System Metrics are statistics around hardware usage, such as GPU
+utilization and network throughput. These metrics are useful for seeing
+whether training is using the hardware effectively. When the System
+Metrics reported for an experiment are below what is expected from the
+hardware, that is a sign that the software may be able to be optimized
+to make better use of the hardware resources.
+
+Specifically, Determined tracks:
+
+-  GPU utilization
+-  GPU memory usage
```

Comment:
should we say free mem here and below? or like `GPU memory usage (as the amount of free GPU memory)`

Reply:
I updated it to be more accurate - "GPU free memory" and "Host available memory"